aerogear/mobile-security-service
# Add the endpoints to update 1 or many versions of the app or disable all +  Replacing all asserts for testing in the service tests layer + more tests + remove asserts

## Motivation
https://issues.jboss.org/browse/AEROGEAR-8356
https://issues.jboss.org/browse/AEROGEAR-8611
https://issues.jboss.org/browse/AEROGEAR-8612

## What
- Disable all versions of an app. 
- Update 1 or many versions of the app 
- Replacing all asserts for testing in the service tests layer
- Improve and increase the tests in the layers
- Remove the `AppPostgreSQLRepository` and `PostgreSQLRepositor` interfaces from pkg/web/apps/apps_psql_repository.go since they are not needed. The `NewPostgreSQLRepository` func uses the generic `Repository` interface.
- Remove the unuseful mock from apps_psql_repository since it is not used. 

## Why
- To attend the UI/project actions requirements
- To attend project definitions

## How
- adding endpoints
- adding tests
- adding service and repository layer implementations



## Verification Steps
- Verify if it passed in all tests
- Verify that the % of tests was increased
- Verify that the project is no longer using asserts
- The unit-test are covering the points, however, please feel free to test it manually as follows. 

#### Setup project and database to do the tests.
- Following the steps described [here](https://github.com/aerogear/mobile-security-service#131-by-using-the-docker-image-of-this-project) in the README
- Then after starting the server, you can create the mock database
- Use can use the [seed] file to do it. See [here](https://github.com/aerogear/mobile-security-service/blob/master/pkg/helpers/db_seed.go)

#### Check endpoint to update all versions by POSTMAN
- Call the endpoint GET `apps` in order to get a valid id
- Call the endpoint POST `/apps/:id/versions/disable` with a JSON with the version struct or empty as the following examples.

```json
{"disabledMessage": "text"}
```

<img width="1139" alt="screenshot 2019-02-28 at 19 29 33" src="https://user-images.githubusercontent.com/7708031/53594174-30b5de00-3b92-11e9-8247-01bfce35cc1c.png">
<img width="1154" alt="screenshot 2019-02-28 at 19 29 11" src="https://user-images.githubusercontent.com/7708031/53594176-30b5de00-3b92-11e9-989d-1a29ef18f2b0.png">


- Check that all versions of the app with ID were updated with the message informed and with the disable as true. 
- The endpoint should return HTTP code 200

#### Check endpoint to update 1 or N versions by POSTMAN
- Call the endpoint GET `apps` in order to get a valid id
- Call the endpoint GET `apps/:id` to get the version struct
- Call the endpoint PUT `/apps/:id/versions` with a JSON with the []versions ( list of versions ).

```json
[
    {
        "id": "9bc87235-6bcb-40ab-993c-8722d86e2201",
        "disabled": true,
        "disabledMessage": "test message"
    },
    {
        "id": "f6fe70a3-8c99-429c-8c77-a2efa7d0b458",
        "disabled": true,
        "disabledMessage": "test message"
    }
]
```

<img width="1316" alt="screenshot 2019-02-28 at 20 45 13" src="https://user-images.githubusercontent.com/7708031/53597336-cef97200-3b99-11e9-9a2a-e6b592d20d20.png">

<img width="1102" alt="screenshot 2019-02-28 at 20 45 01" src="https://user-images.githubusercontent.com/7708031/53597337-cef97200-3b99-11e9-99a3-33fc66601b5c.png">

- Check that all versions sent in the JSON data were updated. 
- The endpoint should return HTTP code 200

## Checklist:


## Progress



## Additional Notes
